src/deepqmc/fit.py

```diff
@@ -187,6 +191,8 @@ def train_step(rng, smpl_state, params, opt_state):
         params, opt_state, E_loc, stats = _step(
             rng_kfac, smpl_state['wf'], params, opt_state, (r, weight)
         )
+        if opt is not None:
```

Comment:
It's not obvious why we update the sampler here, so perhaps this comment is useful?
```suggestion
        if opt is not None:
            # WF was changed in _step, update psi values stored in smpl_state
```